Document style on naming env vars (#1300)
python/devguide: developer-workflow/stdlib.rst

:ref:`contributor agreements <contributor_agreement>`.

.. _issue tracker: https://github.com/python/cpython/issues

> Adding a new env var
> --------------------

Reviewer: ```suggestion
Adding a new environment variable
---------------------------------
```

> From Python 3.13 onwards, name environment variables with underscores for
> readability and accessibility.

Reviewer: I initially failed to parse ["name" as a verb](https://en.wikipedia.org/wiki/Stroop_effect).  I suggest rephrasing.
```suggestion
From Python 3.13 onwards, environment variables name should use
underscores for readability and accessibility.
```

This sentence also assumes the reader is aware that the names are already uppercase, but it might be worth mentioning it explicitly:
```suggestion
Environment variables name should always be uppercase and, from Python 3.13 onwards,
use underscores for readability and accessibility.
```

Author: Good idea, I started off with something like yours, and tried to be clever and make it shorter but made it less clear 🙃

How about something like this?

```suggestion
Environment variables should be uppercase, and from Python 3.13 onwards, use
underscores for readability and accessibility.
```

Reviewer: Technically it's the names -- not the variables themselves -- that are uppercase, but the sentence is clear even if "names" is dropped.  Dropping "always" is also ok.
In my suggestion I was going for a [parenthetical expression](https://en.wikipedia.org/wiki/Parenthesis_(rhetoric)), hence the comma after the "and".
```suggestion
Environment variables should be uppercase and, from Python 3.13 onwards,
use underscores for readability and accessibility.
```
or (with "names" added back):
```suggestion
Environment variables names should be uppercase and, from Python 3.13 onwards,
use underscores for readability and accessibility.
```
Not quite sure if a [possessive apostrophe](https://en.wikipedia.org/wiki/Apostrophe#Possessive_apostrophe) is needed here ("variables' names") -- @CAM-Gerlach?

Reviewer 2: "Names of environment variables" is probably better

> readability and accessibility.

Author: Thanks both, I've combined those into:
```
Names of environment variables should be uppercase and, from Python 3.13
onwards, use underscores for readability and accessibility.
```

> 
> For example, use ``PYTHON_CPU_COUNT`` instead of ``PYTHONCPUCOUNT``.

Reviewer: ```suggestion
For example, use ``PYTHON_CPU_COUNT`` instead of ``PYTHONCPUCOUNT``.
```
Not sure this deserves a separate paragraph.

Author: I think it's clearer like this. If we expand the section, we can adjust as needed.

See also:

* :ref:`python:using-on-envvars`

> * `Discourse discussion
>   <https://discuss.python.org/t/change-environment-variable-style/35180>`__

Reviewer: ```suggestion
* `"Change environment variable style" Discourse discussion
  <https://discuss.python.org/t/change-environment-variable-style/35180>`__
```

Even if in context it's understandable, it's better to avoid generic link text.
